ls1intum/Artemis
# `Exam mode`: Fix an issue when submitting complaints

### Checklist
#### General
#### Server

### Motivation and Context
#6875 fixed a special case where complaints were not possible but missed that check on the server side.

### Description
Fix the server side of this check

### Steps for Testing
Prerequisites:
- 1 Instructor
- 2 Student
- 1 Exam

1. Edit the course to have some low complaint text limit (e.g 10 characters)
2. Submit a exam complaint which uses more than 10 characters
3. Everything should work
4. Course complains should still be limited to 10 characters.
5. Also test exam complaints with course complaints disabled.

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ComplaintResponseService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5222/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ComplaintResponseService.html) | 90% | ✅ |
| [ComplaintService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5222/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ComplaintService.html) | 85% | ✅ |
